test/js/web/websocket/websocket-client.test.ts

```diff
@@ -0,0 +1,273 @@
+import { describe, it, expect, afterEach } from "bun:test";
+import type { Server, Subprocess, WebSocketHandler } from "bun";
+import { serve, spawn } from "bun";
+import { bunEnv, nodeExe } from "harness";
+
+const strings = [
+  {
+    label: "string (ascii)",
+    message: "ascii",
+    bytes: [0x61, 0x73, 0x63, 0x69, 0x69],
+  },
+  {
+    label: "string (latin1)",
+    message: "latin1-©",
+    bytes: [0x6c, 0x61, 0x74, 0x69, 0x6e, 0x31, 0x2d, 0xc2, 0xa9],
+  },
+  {
+    label: "string (utf-8)",
+    message: "utf8-😶",
+    bytes: [0x75, 0x74, 0x66, 0x38, 0x2d, 0xf0, 0x9f, 0x98, 0xb6],
+  },
+];
+
+const buffers = [
+  {
+    label: "Uint8Array (utf-8)",
+    message: new TextEncoder().encode("utf8-🙂"),
+    bytes: [0x75, 0x74, 0x66, 0x38, 0x2d, 0xf0, 0x9f, 0x99, 0x82],
+  },
+  {
+    label: "ArrayBuffer (utf-8)",
+    message: new TextEncoder().encode("utf8-🙃").buffer,
+    bytes: [0x75, 0x74, 0x66, 0x38, 0x2d, 0xf0, 0x9f, 0x99, 0x83],
+  },
+  {
+    label: "Buffer (utf-8)",
+    message: Buffer.from("utf8-🤩"),
+    bytes: [0x75, 0x74, 0x66, 0x38, 0x2d, 0xf0, 0x9f, 0xa4, 0xa9],
+  },
+];
+
+const messages = [...strings, ...buffers];
+
+const binaryTypes = [
+  {
+    label: "nodebuffer",
+    type: Buffer,
+  },
+  {
+    label: "arraybuffer",
+    type: ArrayBuffer,
+  },
+] as const;
+
+let servers: Subprocess[] = [];
+let clients: WebSocket[] = [];
+
+afterEach(() => {
```

Comment:
This afterEach will definitely cause tests to hang due to Subprocess or WebSocket keeping the event loop alive